### PR TITLE
37 ratio of oer content

### PR DESCRIPTION
**Pull request by RMeissnerCC**
Related to KBMBF-502 und SD_WLO-238

### CODE REVIEW

**Review by MRuecklCC — APPROVED** [review]
Just some comments, I'll leave it to you what you fix now or later :-)

File: src/app/api/analytics/storage.py
Comment (RMeissnerCC) [reply src/app/api/analytics/storage.py:39]: Yes, my idea was to move this soon, e.g., to the app or into something more robust. "luckily" currently there is only one function which writes to it and several who read from it. I already see breaking issues there.
I have time until tuesday before we can merge stuff together - I could therefore focus on this issue.

File: src/app/api/analytics/storage.py
Comment (RMeissnerCC) [reply src/app/api/analytics/storage.py:20]: Will do. I fear what might break :D

File: src/app/api/analytics/analytics.py
Comment (RMeissnerCC) [reply src/app/api/analytics/analytics.py:46]: Good point - removed, also for score models

File: src/app/api/analytics/stats.py
Comment (RMeissnerCC) [reply src/app/api/analytics/stats.py:145]: Mmmh, I would confuse CamelCase stuff with classes and not think about type hints. But maybe I just have to get used to that. Changed.

File: src/app/api/analytics/stats.py
Comment (RMeissnerCC) [reply src/app/api/analytics/stats.py:184]: Yes - both are UUIDs. But somehow I cannot compare them directly and there is no function like UUID.isequal(). I am uncertain what it checks with == but it does not work. I stumbled over this issue for half an hour or so

File: src/app/api/analytics/storage.py
Comment (RMeissnerCC) [reply src/app/api/analytics/storage.py:23]: Good spotting - changed!

File: src/app/api/collections/missing_materials.py
Comment (RMeissnerCC) [reply src/app/api/collections/missing_materials.py:191]: I was wondering if this would annoy you ;) Fixed.

File: src/app/api/quality_matrix/quality_matrix.py
Comment (RMeissnerCC) [reply src/app/api/quality_matrix/quality_matrix.py:141]: I want to refactor all these XYZAttributes (elasticsearch, collection, learningmaterials) into one soon. This would be part of it, too, yes. Is it alright if I push it to another PR?

File: src/app/api/score/score.py
Comment (RMeissnerCC) [reply src/app/api/score/score.py:135]: Yes. Agreed.

File: tests/test_stats.py
Comment (RMeissnerCC) [reply tests/test_stats.py:66]: I totally agree,the testing is a pain in the ass.

File: src/app/api/analytics/background_task.py
Comment (RMeissnerCC) [reply src/app/api/analytics/background_task.py:120]: Aah, good spotting. This could resolve the string issue. I check that now.

Comment (RMeissnerCC) [reply src/app/api/analytics/background_task.py:120]: Ay, you are right - the issue is that I use str in Collection class - I added a todo there. Refactoring it now might blow up this PR, since that class is used a lot.

File: src/app/api/analytics/background_task.py
Comment (RMeissnerCC) [reply src/app/api/analytics/background_task.py:121]: I added it and added also a TODO what needs to be done to simplify COLLECTION_NAME_TO_ID alltogether.

File: src/app/api/analytics/stats.py
Comment (MRuecklCC) [reply src/app/api/analytics/stats.py:184]: I think one of them was a string, which is why it did not work. Quick check in python terminal:
```python

u1 = uuid.UUID('12345678123456781234567812345678')
u2 = uuid.UUID('12345678123456781234567812345678')
u1 == u2
> True
```
so comparing them should be no problem.

Comment (RMeissnerCC) [reply src/app/api/analytics/stats.py:184]: Yes, one of them was a string.

File: src/app/api/analytics/storage.py
Comment (RMeissnerCC) [reply src/app/api/analytics/storage.py:24]: Yes, it is the same.I will refactor

Comment (RMeissnerCC) [reply src/app/api/analytics/storage.py]: Done

**Comment by MRuecklCC** [comment]
:+1:

**Pull request merged**
